week02/49. Group Anagrams.cpp

```diff
@@ -0,0 +1,20 @@
+class Solution {
+public:
+      // TC: O(NlogN) for the for loop O(n)+for sort O(nlogn)
+       // MC: O(1)
+    vector<vector<string>> groupAnagrams(vector<string>& strs) {
+        vector<vector<string>>ans;
+        unordered_map<string,vector<string>>umap;
+        for(auto x:strs)
+        {
+            string temp = x;
+            sort(x.begin(),x.end());
+            umap[x].push_back(temp);
+        }

```

Comment:
```suggestion
        for(auto str : strs)
        {
            string sorted = str;
            sort(sorted.begin(), sorted.end());
            umap[sorted].push_back(str);
        }

```

Doesn't it look better?